### PR TITLE
packaging: Fix sev cache kernel issue

### PR DESCRIPTION
**Pull request by UnmeshDeodhar**
Fix cache kernel issue introduced by #6177

Fixes: #6362

### CODE REVIEW

**Comment by fidencio** [comment]
@UnmeshDeodhar, can I see the logs with the errors you're facing?

**Comment by UnmeshDeodhar** [comment]
@fidencio Here is the log from the cache kernel job: http://jenkins.katacontainers.io/view/Cached%20jobs/job/kata-containers-2.0-kernel-sev-cc-x86_64-DEBUG/27/console

**Comment by fidencio** [comment]
> @fidencio Here is the log from the cache kernel job: http://jenkins.katacontainers.io/view/Cached%20jobs/job/kata-containers-2.0-kernel-sev-cc-x86_64-DEBUG/27/console

```
18:05:26 quay.io/kata-containers/cc-builders:kernel-e8c63ccc4d93ae1e4019ed1e310938328e8044bb-x86_64
18:05:26 ./tools/packaging/static-build/cache_components.sh: line 50: kernel_version: unbound variable
18:05:26 Build step 'Execute shell' marked build as failure
```

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh [hunk 1]
```diff
@@ -400,7 +404,7 @@ install_cached_kernel_component() {
 	kernel_version="${2}"
 	module_dir="${3:-}"
 
-	install_cached_compnent \
+	install_cached_component \
```
Comment (fidencio) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:407]: This change is not related to the PR.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh [hunk 2]
```diff
@@ -17,6 +17,10 @@ readonly script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
 source "${script_dir}/../../scripts/lib.sh"
 
+readonly packaging_root_dir="${repo_root_dir}/tools/packaging"
+patches_path=""
+readonly default_patches_dir="${packaging_root_dir}/kernel/patches"
```
Comment (fidencio) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:22]: Imagine we're calling `get_config_version` in different files, and then we'd have to set this up in each one of the calls.

This makes me think that here is definitely not the place to add set this variables.

File: tools/packaging/static-build/cache_components.sh [hunk 3]
```diff
@@ -16,6 +16,10 @@ export KATA_BUILD_CC="${KATA_BUILD_CC:-}"
 export TEE="${TEE:-}"
 export FIRMWARE="${FIRMWARE:-}"
 
+readonly packaging_root_dir="${repo_root_dir}/tools/packaging"
+patches_path=""
+readonly default_patches_dir="${packaging_root_dir}/kernel/patches"
```
Comment (fidencio) [inline tools/packaging/static-build/cache_components.sh:21]: Like adding this here again. :-)

**Review by fidencio — CHANGES_REQUESTED** [review]
Okay, now I got why you were setting up those, and I must admit I totally missed the `get_config_version` call.

Anyways, the correct fix is, instead of setting those var on every single place you call `get_config_version`, you make sure those are correct setup from within that function.

I think this is the correct fix:
````
⋊> kata-containers on CCv0 ≡ ⨯ git diff
diff --git a/tools/packaging/guest-image/build_image.sh b/tools/packaging/guest-image/build_image.sh
index ee77f396b..86c00dd46 100755
--- a/tools/packaging/guest-image/build_image.sh
+++ b/tools/packaging/guest-image/build_image.sh
@@ -20,7 +20,7 @@ source "${packaging_root_dir}/scripts/lib.sh"
 readonly osbuilder_dir="$(cd "${repo_root_dir}/tools/osbuilder" && pwd)"
 
 patches_path=""
-readonly default_patches_dir="${packaging_root_dir}/kernel/patches"
+readonly default_patches_dir="${repo_root_dir}/tools/packaging/kernel/patches"
 
 export GOPATH=${GOPATH:-${HOME}/go}
 
diff --git a/tools/packaging/scripts/lib.sh b/tools/packaging/scripts/lib.sh
index ef74acad3..de136312c 100755
--- a/tools/packaging/scripts/lib.sh
+++ b/tools/packaging/scripts/lib.sh
@@ -115,7 +115,7 @@ get_config_and_patches() {
 
 get_config_version() {
        get_config_and_patches
-       config_version_file="${default_patches_dir}/../kata_config_version"
+       config_version_file="${repo_root_dir}/tools/packaging/kernel/kata_config_version"
        if [ -f "${config_version_file}" ]; then
                cat "${config_version_file}"
        else
````

The changes related kernel_version -> current_kernel_version are good.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh (hunk 1 above)
Comment (fidencio) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:407]: My bad, it totally is. :-)

**Review by fidencio — APPROVED** [review]
lgtm, thanks @UnmeshDeodhar!

**Review by GabyCT — APPROVED** [review]

**Comment by UnmeshDeodhar** [comment]
@GabyCT @fidencio fyi, all tests are passing except tdx. We except significant time gains with this PR.

**Comment by fidencio** [comment]
http://10.112.240.208:8080/job/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/730/ passed but didn't report back, we can get this one merged.

**Pull request merged**